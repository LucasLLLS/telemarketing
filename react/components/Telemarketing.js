import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { intlShape } from 'react-intl'
import { withRuntimeContext } from 'render'

import LoginAsCustomer from './LoginAsCustomer'
import LogoutCustomerSession from './LogoutCustomerSession'
import TelemarketingIcon from '../icons/TelemarketingIcon'

import translate from '../utils/translate'
import { clientPropTypes } from '../utils/propTypes'

/** Telemarketing render component */
class Telemarketing extends Component {
  render() {
    const {
      intl,
      client,
      loading,
      emailInput,
      onInputChange,
      onSetSession,
      onDepersonify,
      attendantEmail,
      runtime: { hints: { mobile } }
    } = this.props

    const isLogged = client

    return (
      <div
        className={`vtex-telemarketing tc white h2 flex justify-between w-100 f7 ${
          isLogged ? 'bg-red' : 'bg-black-90'
          } z-999 pa2`}
      >
        <div className="flex items-center">
          <TelemarketingIcon />

          {!mobile && (
            <div className="ml2">
              {translate('telemarketing.attendant', intl)}
              <b>{`: ${attendantEmail}`}</b>
            </div>
          )}
        </div>
        {isLogged ? (
          <LogoutCustomerSession
            intl={intl}
            client={client}
            loading={loading}
            onDepersonify={onDepersonify}
            attendantEmail={attendantEmail}
          />
        ) : (
            <LoginAsCustomer
              intl={intl}
              loading={loading}
              emailInput={emailInput}
              onInputChange={onInputChange}
              onSetSession={onSetSession}
              attendantEmail={attendantEmail}
            />
          )}
      </div>
    )
  }
}

Telemarketing.propTypes = {
  /** Intl object */
  intl: intlShape.isRequired,
  /** Impersonated customer info */
  client: clientPropTypes,
  /** Loading status */
  loading: PropTypes.bool.isRequired,
  /** Email input value */
  emailInput: PropTypes.string.isRequired,
  /** Attendant email */
  attendantEmail: PropTypes.string.isRequired,
  /** Function to set the session */
  onSetSession: PropTypes.func.isRequired,
  /** Function to set the emailInput value */
  onInputChange: PropTypes.func.isRequired,
  /** Function to depersonify the impersonated customer */
  onDepersonify: PropTypes.func.isRequired,
}
export default withRuntimeContext(Telemarketing)
