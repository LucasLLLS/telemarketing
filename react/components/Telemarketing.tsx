import React, { ReactNode } from 'react'
import { Container } from 'vtex.store-components'
import { IconAssistantSales } from 'vtex.store-icons'
import classnames from 'classnames'
import useDevice from 'vtex.device-detector/useDevice'

import LoginAsCustomer from './LoginAsCustomer'
import LogoutCustomerSession from './LogoutCustomerSession'
import styles from '../telemarketing.css'
import { FormattedMessage } from 'react-intl'

interface Props {
  /** Attendant email */
  attendantEmail: string
  /** Impersonated customer info */
  client?: Client
  /** Email input value */
  emailInput: string
  /** Loading status */
  loading: boolean
  /** Function to depersonify the impersonated customer */
  onDepersonify: () => any
  /** Function to set the emailInput value */
  onInputChange: (s: string) => void
  /** Function to impersonate */
  onImpersonate: (s: string) => void
  /** Children */
  readonly children?: ReactNode
}

/** Telemarketing render component */
const Telemarketing = ({
  client,
  loading,
  emailInput,
  onInputChange,
  onImpersonate,
  onDepersonify,
  attendantEmail,
}: Props) => {
  const { isMobile } = useDevice()

  const containerClasses = classnames(
    styles.container,
    'flex justify-center tc h2 t-mini pa2 w-100'
  )

  return (
    <div className={classnames('w-100', styles.wrapper, !!client ? 'bg-emphasis c-on-emphasis' : 'bg-base--inverted c-on-base--inverted')}>
      <Container className={containerClasses}>
        <div className="flex justify-between w-100 mw9">
          <div className="flex pl3 w-50 items-center c-on-base--inverted">
            <IconAssistantSales />
            <div className="ml2">
              {isMobile ? (
                <span>
                  {attendantEmail.slice(0, attendantEmail.indexOf('@'))}
                </span>
              ) : (
                <FormattedMessage
                  id="store/telemarketing.attendant"
                  values={{ attendant: attendantEmail }} />
              )}
            </div>
          </div>
          {!!client ? (
            <LogoutCustomerSession
              client={client}
              loading={loading}
              onDepersonify={onDepersonify}
              attendantEmail={attendantEmail}
              mobile={isMobile}
            />
          ) : (
            <LoginAsCustomer
              loading={loading}
              emailInput={emailInput}
              onInputChange={onInputChange}
              onImpersonate={onImpersonate}
              attendantEmail={attendantEmail}
              mobile={isMobile}
            />
          )}
        </div>
      </Container>
    </div>
  )
}

export default Telemarketing
