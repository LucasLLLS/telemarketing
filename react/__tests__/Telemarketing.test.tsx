import React from 'react'
import { injectIntl } from 'react-intl'
import { render } from 'react-testing-library'

import Telemarketing from '../components/Telemarketing'

describe('<Telemarketing /> component', () => {
  it('should match snapshot', () => {
    const component = render(injectIntl(<Telemarketing />)).asFragment()
    expect(component).toMatchSnapshot()
  })
})
