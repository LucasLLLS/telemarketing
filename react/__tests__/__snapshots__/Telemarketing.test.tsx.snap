// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Telemarketing /> component should match snapshot 1`] = `<DocumentFragment />`;
